src/crypto/sha256.cpp

```diff
@@ -11,11 +11,13 @@
 
 #if defined(__x86_64__) || defined(__amd64__)
 #include <cpuid.h>
```

Comment:
Nit: no need to risk including the not-guaranteed-to-exist header. Move the ```#if``` up a bit?